varnishgather

```diff
@@ -428,6 +436,21 @@ done
 
 shift $((OPTIND - 1))
 
+if [ -z "$NAME" ]; then
+    INSTANCE_NAME=$(get_varnishd_instance)
+    [ -n "$INSTANCE_NAME" ] && NAME="-n $INSTANCE_NAME"
+else
+    INSTANCE_NAME="${NAME#-n }"
+fi
+
+if [ -n "$NAME" ] && ! validate_instance; then
+    INSTANCE_NAME=$(get_varnishd_instance)
+    [ -n "$INSTANCE_NAME" ] && NAME="-n $INSTANCE_NAME" || NAME=""
+fi
```

Comment:
Pål wrote:

> If there is a -n argument, but varnishadm ping -n does not work, report that and suggest other -n values that might be better, but run the rest of the script.

I don't think this code is doing that

Reply:
you are right, I did not go back to your original comments I focused on fixing my laziness before updating.  I'll work make sure to add a varnishadm ping check in here, my bad

Reply:
after re-reading your original statement.  I completely skipped over what you wanted.  I focused on just ensuring the name was included in the script execution with or without the client adding it, or even if they add the wrong name it picks the proper name.  A new day and a revisit to your comment Pal has me feeling you would rather have the following.

1. script executed, check provided -n value (if correct it goes on about it's marry way)
2. script executed, no -n provided but there is a name, so we spit out to the client they should rerun the script with -n $propername (exit script)
3.. script executed with -n and wrong name, return the potential proper names we gather and present that to the client with the NAME's we found with echo statements to run the script properly but then exit script.

I went about this task thinking I should just do the -n work for them.  There is a potential for compromise here.  I could go do the -n capture and then also, print out to the client that we had to do this because they failed to provide the information.  Even warning we chose the first -n process we found, therefore if they meant X/Y/Z then they should re-run the script with the proper name?

Reply:
to be completely exhaustive, I think this is the proper logic:
- `-n` given
  - `varnishadm ping` works
    - proceed as usual
  - `varnishadm ping` doesn't work
    - warn that the name may be wrong, print out a list of candidates if any, then proceed
- no `-n`
  - `varnishadm ping` works
    - proceed as usual
  - if there's exactly one candidate
    - warn that we defaulted to that auto-selected name, proceed if `varnishadm ping` works, error out otherwise
  - there is 0 candidate
    - warn that `varnishd` may not be running
  - there are more than 1 candidate
    - list the candidates, error out

@zackmay , @hermunn , how does it look?